api/src/main/java/org/pmiops/workbench/billing/InitialCreditsService.java

```diff
@@ -242,6 +258,155 @@ public double getWorkspaceCreatorFreeCreditsRemaining(DbWorkspace dbWorkspace) {
     return Math.max(creatorFreeCreditsRemaining, 0);
   }
 
+  public void checkCreditsExpirationForUserIDs(List<Long> userIdsList) {
+    if (userIdsList != null && !userIdsList.isEmpty()) {
+      Iterable<DbUser> users = userDao.findAllById(userIdsList);
+      users.forEach(this::checkExpiration);
+    }
+  }
+
+  public Optional<Timestamp> getCreditsExpiration(DbUser user) {
+    return Optional.ofNullable(user.getUserInitialCreditsExpiration())
+        .filter(exp -> !exp.isBypassed()) // If the expiration is bypassed, return empty.
+        .filter(exp -> !institutionService.shouldBypassForCreditsExpiration(user))
+        .map(DbUserInitialCreditsExpiration::getExpirationTime);
+  }
+
+  public boolean haveCreditsExpired(DbUser user) {
+    return getCreditsExpiration(user)
+        .map(expirationTime -> !expirationTime.after(clockNow()))
+        .orElse(false);
+  }
+
+  // Returns true if the user's credits are expiring within the initialCreditsExpirationWarningDays.
+  private boolean areCreditsExpiringSoon(DbUser user) {
```

Comment:
naming question: 
`isCredit` or 'areCredits'

also want to hear @jmthibault79 's suggestion

Reply:
I believe that both are grammatically correct, however since most, if not all, other references use the plural (credits), I would prefer `areCreditsExpiringSoon`.